src/coreclr/vm/jithelpers.cpp

```diff
@@ -331,10 +331,9 @@ HCIMPLEND
 // the most optimal TLS access pattern for the platform when inlining the
 // GetThreadLocalStaticBaseIfExistsAndInitialized function
```

Comment:
IIRC, C/C++ compilers had code quality issues with thread statics and this exact definition was required to get the optimal codegen for performance critical thread statics like this one. It is quite possible that these codegen issues have been fixed in C/C++ compilers and we do not need to worry about it anymore. You may want to double check the C++ codegen before/after this change to make sure that it is not introducing a regression.

Reply:
Checked the codegen and there wasn't any obvious differences between the two. However, there was a difference related to the linkage issue mentioned at https://github.com/dotnet/runtime/pull/114660#issuecomment-2803384274. I changed this back and clarified we are using the compiler specific due to linkage issues.